tests/prague/eip7623_increase_calldata_cost/test_transaction_validity.py

```diff
@@ -0,0 +1,524 @@
+    """
+    return None
+
+
+@pytest.fixture
+def contract_creating_tx(to: Address | None) -> bool:
+    """
+    Gas delta for the transaction, used to generate an invalid transaction.
+    """
+    return to is None
+
+
+@pytest.fixture
+def tx_data(
+    fork: Fork,
+    data_test_type: DataTestType,
+    access_list: List[AccessList] | None,
+    authorization_list: List[AuthorizationTuple] | None,
+    contract_creating_tx: bool,
+) -> Bytes:
+    """
+    All tests in this file use data that is generated dynamically depending on the case and the
+    attributes of the transaction in order to reach the edge cases where the floor gas cost is
+    equal or barely greater than the intrinsic gas cost.
+
+    We have two different types of tests:
+    - FLOOR_GAS_COST_LESS_THAN_OR_EQUAL_TO_INTRINSIC_GAS: The floor gas cost is less than or equal
+        to the intrinsic gas cost, which means that the size of the tokens in the data are not
+        enough to trigger the floor gas cost.
+    - FLOOR_GAS_COST_GREATER_THAN_INTRINSIC_GAS: The floor gas cost is greater than the intrinsic
+        gas cost, which means that the size of the tokens in the data are enough to trigger the
+        floor gas cost.
+    """
+
+    def tokens_to_data(tokens: int) -> Bytes:
+        return Bytes(b"\x01" * (tokens // 4) + b"\x00" * (tokens % 4))
+
+    # Start with zero data and check the difference in the gas calculator between the
+    # intrinsic gas cost and the floor gas cost.
+    fork_intrinsic_cost_calculator = fork.transaction_intrinsic_cost_calculator()
+
+    def transaction_intrinsic_cost_calculator(tokens: int) -> int:
+        return fork_intrinsic_cost_calculator(
+            calldata=tokens_to_data(tokens),
+            contract_creation=contract_creating_tx,
+            access_list=access_list,
+            authorization_list_or_count=authorization_list,
+            return_cost_deducted_prior_execution=True,
+        )
+
+    fork_data_floor_cost_calculator = fork.transaction_data_floor_cost_calculator()
+
+    def transaction_data_floor_cost_calculator(tokens: int) -> int:
+        return fork_data_floor_cost_calculator(data=tokens_to_data(tokens))
+
+    if transaction_data_floor_cost_calculator(0) >= transaction_intrinsic_cost_calculator(0):
+        # Special case which is a transaction with no extra intrinsic gas costs other than the
+        # data cost.
+        if data_test_type == DataTestType.FLOOR_GAS_COST_LESS_THAN_OR_EQUAL_TO_INTRINSIC_GAS:
+            return Bytes(b"")
+        else:
+            return Bytes(b"\0")
+
+    # Start with 1000 tokens and if the intrinsic gas cost is greater than the floor gas cost,
+    # multiply the number of tokens by 2 until it's not.
+    tokens = 1000
+    while transaction_data_floor_cost_calculator(tokens) < transaction_intrinsic_cost_calculator(
+        tokens
+    ):
+        tokens *= 2
+
+    # Binary search to find the minimum number of tokens that will trigger the floor gas cost.
```

Comment:
thats exactly how in one of the tests Ori did trials to find out and trigger the OOG scenarios

Reply:
Yes, the idea is that even if the gas costs change in a future fork this test case should still be able to generate edge cases without intervention.